### ISSUE TITLE
Can you please explain how the network learns  prototypes 

### ISSUE THREAD

**Issue by fangy123434** [fanconic/this-does-not-look-like-that#3]
After the training is completed, I understand the process. However, I cannot understand how the prototypes are learned at the beginning? Is it user defined? The paper doesn't say so.

**Comment by fanconic** [comment]
Are you now generally referring to how Prototypes are learned, or in our specific experiments to show some shortcomings in the interpretability mechanism?

If you are generally unsure of the process, I would recommend reading the original paper from Chen et al.: https://arxiv.org/pdf/1806.10574.pdf. Here they explain how the prototypes are learned.

In the beginning, the prototypes are randomly placed and then learned in the "push" face of the training.

**Comment by fangy123434** [comment]
> Are you now generally referring to how Prototypes are learned, or in our specific experiments to show some shortcomings in the interpretability mechanism?
> 
> If you are generally unsure of the process, I would recommend reading the original paper from Chen et al.: https://arxiv.org/pdf/1806.10574.pdf. Here they explain 

> Are you now generally referring to how Prototypes are learned, or in our specific experiments to show some shortcomings in the interpretability mechanism?
> 
> If you are generally unsure of the process, I would recommend reading the original paper from Chen et al.: https://arxiv.org/pdf/1806.10574.pdf. Here they explain how the prototypes are learned.
> 
> In the beginning, the prototypes are randomly placed and then learned in the "push" face of the training.

Thank you for your reply. I have read the original paper, but I still don’t understand how prototype is learned. Is it through the projection of the second phase of the training algorithm in the original paper?

**Comment by fanconic** [comment]
The prototypes themselves are learned in the first phase (In the original Paper, Section 2.2 and subsection "Stochastic gradient descent (SGD) of layers before last layer"). After they have been learned, these prototypes are "pushed" towards the latent space representation of training samples. Therefore the prototypes become similar to actual training patches (Section 2.2 Paper, subsection "Projection of prototypes").

You might most likely get a better explanation by contacting the authors from the original papers.

**Comment by fangy123434** [comment]
> 原型本身是在第一阶段学习的（在原始论文的第 2.2 节和“最后一层之前的层的随机梯度下降（SGD）”小节）。在学习之后，这些原型被“推向”训练样本的潜在空间表示。因此，原型变得类似于实际的训练补丁（第 2.2 节论文，“原型投影”小节）。
> 
> 通过联系原始论文的作者，您很可能会得到更好的解释。

Thank you for your reply and wish you a happy life.

**Comment by fanconic** [comment]
谢谢你，我希望这对你有帮助!
